src/utilities/sql/SqlFile.hpp

```diff
@@ -35,6 +35,7 @@
 #include "SummaryData.hpp"
 #include "SqlFileDataDictionary.hpp"
 #include "SqlFileEnums.hpp"
+#include "SqlFile_Impl.hpp"
```

Comment:
That's the thing I dislike the most. Because I use variadic templates (to forward to the Impl variadic functions) in this file, I need to define them in the header file. And therefore I must include the SqlFile_Impl.hpp becase otherwise `m_impl` is not a complete type with just a forward declaration...

Reply:
well the Impls are breaking through all over the place as a matter of practice, because of how casting works. I'm wondering more about how these variadic templates look to swig. Have you evaluated this api from the Ruby point of view?